### PR TITLE
Fix a bug in Tree leaf featurizer entry point, and add a test for it.

### LINKED ISSUE #130: TreeEnsembleFeaturizer entry point doesn't instantiate the scorer correctly
It should instantiate the scorer on top of the transformed data, but instead it uses the input data.

### PR DESCRIPTION
**Pull request by yaeldekel**
The tree leaf featurizer entry point doesn't pass the correct data to the scorer it creates - it should pass the transformed data, but instead it passes the input data.
Fixes #130

### CODE REVIEW

File: test/Microsoft.ML.Core.Tests/UnitTests/TestEntryPoints.cs [hunk 1]
```diff
@@ -2521,5 +2521,56 @@ public void EntryPointPrepareLabelConvertPredictedLabel()
...
+            var view = treeLeaf.OutputData;
+            Assert.True(view.Schema.TryGetColumnIndex("Trees", out int col));
+            VBuffer<float> features = default(VBuffer<float>);
+            using (var curs = view.GetRowCursor(c => c == col))
+            {
+                var getter = curs.GetGetter<VBuffer<float>>(col);
+                while (curs.MoveNext())
+                {
+                    getter(ref features);
+                    Assert.True(features.Count > 0);
```
Comment (TomFinley) [inline test/Microsoft.ML.Core.Tests/UnitTests/TestEntryPoints.cs]: >features.Count [](start = 32, length = 14)

Can we make a few more definite statements about this vector to make the test a bit stronger? For example: It should not only have some non-zero entries, we'd expect it to be a particular length, etc., right?

File: src/Microsoft.ML.FastTree/TreeEnsembleFeaturizer.cs [hunk 2]
```diff
@@ -715,16 +717,16 @@ public static IDataTransform CreateForEntryPoint(IHostEnvironment env, Arguments
                 // be non-null.
                 var vm = predictor as IValueMapper;
                 ch.CheckUserArg(vm != null, nameof(args.PredictorModel), "Predictor does not have compatible type");
-                if (data != null && vm?.InputType.VectorSize != data.Schema.Feature.Type.VectorSize)
+                if (data != null && vm.InputType.VectorSize != data.Schema.Feature.Type.VectorSize)
```
Comment (TomFinley) [inline src/Microsoft.ML.FastTree/TreeEnsembleFeaturizer.cs]: >if (data != null && vm.InputType.VectorSize != data.Schema.Feature.Type.VectorSize) [](start = 16, length = 83)

If someone were to fit in a predictor that did implement `IValueMapper` (which are most), but that is nevertheless not a tree, what do we expect would happen? From my reading of the code it would just apply a generic scorer. Should we at least verify, in some fashion, that it *is* the type of predictor we expect beyond being merely a `IValueMapper`?

Comment (yaeldekel) [reply src/Microsoft.ML.FastTree/TreeEnsembleFeaturizer.cs]: Hi Tom, thanks for reviewing this change. If the predictor is an IValueMapper but is not a tree, then the constructor of TreeEnsembleFeaturizerBindableMapper below will fail.

---
In reply to: [187707032](https://github.com/dotnet/machinelearning/pull/131#discussion_r187707032) [](ancestors = 187707032)

**Comment by glebuk** [comment]
@me is added to the review. #Closed

**Comment by glebuk** [comment]
@glebuk is added to the review. #Closed

File: test/Microsoft.ML.Core.Tests/UnitTests/TestEntryPoints.cs [hunk 3]
```diff
@@ -2521,5 +2521,70 @@ public void EntryPointPrepareLabelConvertPredictedLabel()
...
             }
         }
+
+        [Fact]
+        public void EntryPointTreeLeafFeaturizer()
+        {
+            var dataPath = GetDataPath(@"adult.tiny.with-schema.txt");
+            var inputFile = new SimpleFileHandle(Env, dataPath, false, false);
+            var dataView = ImportTextData.ImportText(Env, new ImportTextData.Input { InputFile = inputFile }).Data;
+            var cat = Categorical.CatTransformDict(Env, new CategoricalTransform.Arguments()
```
Comment (glebuk) [inline test/Microsoft.ML.Core.Tests/UnitTests/TestEntryPoints.cs]: >CatTransformDict [](start = 34, length = 16)

Would it be easier to use the LearningPipeline for the test here?
